app/controllers/publishers/site_banners_controller.rb

```diff
@@ -5,14 +5,75 @@ class Publishers::SiteBannersController < ApplicationController
 
   MAX_IMAGE_SIZE = 10_000_000
 
-  def new
-    @site_banner = current_publisher.site_banner || SiteBanner.new
+  # Fetch Banner by channel_id, if it does not exist create a template banner.
+  def fetch
+
+    site_banner = current_publisher.site_banners.find_by(channel_id: params[:channel_id])
+
+    if site_banner
+      data = site_banner.read_only_react_property
+      data[:backgroundImage] = data[:backgroundUrl]
+      data[:logoImage] = data[:logoUrl]
+      render(json: data.to_json)
+    else
+      site_banner = new(params[:channel_id])
```

Comment:
This isn't idempotent, a GET shouldn't create a new entry in the database similar to what Albert said. I'm going to write something up on a different way to structure this 👍